File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
class Solution {
    private int count = 0;
    private int res = 0;
    public int kthSmallest(TreeNode root, int k) {
        traverse(root,k);
        return res;
    }
    private void traverse(TreeNode node, int k){
        if(node == null) return;
        traverse(node.left,k);
        count++;
        if(count == k){
            res = node.val;
            return;
        }
        traverse(node.right,k);
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.count = 0
        self.res = 0       
        def traverse(node):
            if not node:
                return
            
            # 1. Traverse left subtree
            traverse(node.left)
            
            # 2. Process current node
            self.count += 1
            if self.count == k:
                self.res = node.val
                return  # Optimization: stop early
            
            # 3. Traverse right subtree
            traverse(node.right)
        
        traverse(root)
        return self.res